test/extended/router/gatewayapicontroller.go

```diff
@@ -231,31 +206,48 @@ var _ = g.Describe("[sig-network-edge][OCPFeatureGate:GatewayAPIController][Feat
 	})
 
 	g.It("Ensure HTTPRoute object is created", func() {
+		// skip since gateway needs LB service
+		skipGatewayIfNonCloudPlatform(oc)
+
 		g.By("Ensure default GatewayClass is accepted")
 		errCheck := checkGatewayClass(oc, gatewayClassName)
 		o.Expect(errCheck).NotTo(o.HaveOccurred(), "GatewayClass %q was not installed and accepted", gatewayClassName)
 
 		g.By("Getting the default domain")
 		defaultIngressDomain, err := getDefaultIngressClusterDomainName(oc, time.Minute)
-		o.Expect(err).NotTo(o.HaveOccurred(), "failed to find default domain name")
+		o.Expect(err).NotTo(o.HaveOccurred(), "Failed to find default domain name")
 		customDomain := strings.Replace(defaultIngressDomain, "apps.", "gw-custom.", 1)
 
 		g.By("Create a custom Gateway for the HTTPRoute")
 		gw := names.SimpleNameGenerator.GenerateName("gateway-")
 		gateways = append(gateways, gw)
 		_, gwerr := createAndCheckGateway(oc, gw, gatewayClassName, customDomain)
-		o.Expect(gwerr).NotTo(o.HaveOccurred(), "failed to create Gateway")
+		o.Expect(gwerr).NotTo(o.HaveOccurred(), "Failed to create Gateway")
 
 		g.By("Create the http route using the custom gateway")
-		httproutes = append(httproutes, "test-httproute")
-		defaultRoutename := "test-hostname.gwapi." + customDomain
+		defaultRoutename := "test-hostname." + customDomain
 		createHttpRoute(oc, gw, "test-httproute", defaultRoutename, "echo-pod-"+gw)
 
 		g.By("Checking the http route using the default gateway is accepted")
 		assertHttpRouteSuccessful(oc, gw, "test-httproute")
 	})
 })
 
+func skipGatewayIfNonCloudPlatform(oc *exutil.CLI) {
+	infra, err := oc.AdminConfigClient().ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(infra).NotTo(o.BeNil())
+
+	platformType := infra.Status.PlatformStatus.Type
+	o.Expect(platformType).NotTo(o.BeNil())
```

Comment:
`infra.Status.PlatformStatus` can be `nil` which may cause a `panic` during dereferencing.
```suggestion
	o.Expect(infra.Status.PlatformStatus).NotTo(o.BeNil())
	platformType := infra.Status.PlatformStatus.Type
	o.Expect(platformType).NotTo(o.BeEmpty())
```

Reply:
Good catch! Thanks for the suggestion.